integration-tests/tests/specs/features/file-sharing.spec.ts

```diff
@@ -0,0 +1,110 @@
+import { Page, expect } from '@playwright/test';
+import { test } from '../../fixtures/group.fixture';
+import { BulkSubmissionPage, SingleSequenceSubmissionPage } from '../../pages/submission.page';
+
+test('submit a single sequence with two files', async ({ pageWithGroup, page }) => {
+    test.setTimeout(90000);
+    const submissionPage = new SingleSequenceSubmissionPage(pageWithGroup);
+
+    await submissionPage.navigateToSubmissionPage('Test organism (with files)');
+    await submissionPage.fillSubmissionFormDummyOrganism({
+        submissionId: 'TEST-ID-123',
+        country: 'Uganda',
+        date: '2023-10-15',
+    });
+    let cleanup: () => Promise<void>;
+    try {
+        cleanup = await submissionPage.uploadExternalFiles('raw_reads', {
+            'hello.txt': 'Hello',
+            'world.txt': 'World',
+        });
+
+        await expect(page.getByText('✓').first()).toBeVisible();
+        await expect(page.getByText('✓').nth(1)).toBeVisible();
+    } finally {
+        if (cleanup) {
+            await cleanup();
+        }
+    }
+
+    await submissionPage.acceptTerms();
+    const reviewPage = await submissionPage.submitSequence();
+    await reviewPage.waitForZeroProcessing();
+
+    await reviewPage.releaseValidSequences();
+
+    await page.getByRole('link', { name: 'released sequences' }).click();
+    while (!(await page.getByRole('link', { name: /LOC_/ }).isVisible())) {
+        await page.reload();
+        await page.waitForTimeout(2000);
+    }
```

Comment:
Optionally we could extract this into a function which I suspect would be quite generically useful - I actually think I maybe used to have one but it might have disappeared during some big refactoring

Reply:
I agree. There is still a similar function but it's in search.page.ts (and now here we're not on the search page - even though they are very very similar). One thing I found difficult-ish to generalize, is that this won't work if there will be multiple matching things.

So yes we should probably do it, but I'll leave it for now - maybe in the future it'll be more obvious how to generalize it best. Thanks for bringing it up and making me look into it again!